grpc/src/main/java/com/linecorp/armeria/server/grpc/FramedGrpcService.java

```diff
@@ -319,10 +329,11 @@ private <I, O> void startCall(ServerMethodDefinition<I, O> methodDef, ServiceReq
         call.setListener(listener);
         call.startDeframing();
         ctx.whenRequestCancelling().handle((cancellationCause, unused) -> {
-            Status status = Status.CANCELLED.withCause(cancellationCause);
-            if (cancellationCause instanceof RequestTimeoutException) {
-                status = status.withDescription("Request timed out");
-            }
+            final Status status = call.exceptionHandler()
+                .orElse(cancellationGrpcExceptionHandler)
+                .apply(ctx, cancellationCause, headers);
```

Comment:
Now sure if we should pass the request headers or just empty headers. If this represents response headers, then I suppose it should be empty?